### PR TITLE
Remove references to `pd.Int64Index` in anticipation of deprecation

### PR DESCRIPTION
**Pull request by jsignell**
Ref https://github.com/dask/dask/issues/8036

### CODE REVIEW

**Comment by jsignell** [comment]
I don't think this should cause GPU issues, but pinging @dask/gpu just in case.

**Comment by jsignell** [comment]
rerun tests

**Comment by jsignell** [comment]
This one makes sense to me, but I'd appreciate a second set of eyes @jrbourbeau. It side-steps a very loud `FutureWarning` on pandas that is triggered on the get for `pd.Int64Index`

File: dask/dataframe/tests/test_utils_dataframe.py [hunk 1]
```diff
@@ -205,9 +209,10 @@ def test_meta_nonempty_index():
     assert type(res) is pd.RangeIndex
     assert res.name == idx.name
 
-    idx = pd.Int64Index([1], name="foo")
+    idx = pd.Index([1], name="foo", dtype="int")
     res = meta_nonempty(idx)
-    assert type(res) is pd.Int64Index
+    assert isinstance(res, pd.Index)
```
Comment (jrbourbeau) [inline dask/dataframe/tests/test_utils_dataframe.py:214]: Nitpick: it looks like we've backed off the `is` comparison here in favor of `isinstance`, which is less precise. Maybe `assert type(res) is type(idx)` would be equivalent to what's currently in `main`?

**Review by jrbourbeau — APPROVED** [review]
Thanks @jsignell! This looks good

File: dask/dataframe/tests/test_utils_dataframe.py (hunk 1 above)
Comment (jsignell) [reply dask/dataframe/tests/test_utils_dataframe.py:214]: yes! Good call. I think I had a different solution originally where this was the best I could do.

**Pull request merged**